admin-dev/themes/new-theme/js/pages/product/edit/product-partial-updater.js

```diff
@@ -0,0 +1,160 @@
+
+const {$} = window;
+
+/**
+ * When product is edited we want to send only partial updates
+ * so this class compares the initial data from the form computes
+ * the diff when form is submitted And dynamically build another
+ * form to submit only updated data (along with required fields
+ * token and such).
+ *
+ * It also disabled the submit button as long as no data has been
+ * modified by the user.
+ */
+export default class ProductPartialUpdater {
+  constructor($productForm, $productFormSubmitButton) {
+    this.$productForm = $productForm;
+    this.$productFormSubmitButton = $productFormSubmitButton;
+  }
+
+  /**
+   * This the public method you need to use to start this component
+   * ex: new ProductPartialUpdater($productForm, $productFormSubmitButton).watch();
+   */
+  watch() {
+    this.$productFormSubmitButton.prop('disabled', true);
+    this.initialData = this.getFormDataAsObject();
+    this.$productForm.submit((e) => this.updatePartialForm(e));
+    this.$productForm.on('change', ':input', () => this.updateSubmitButtonState());
+  }
+
+  /**
+   * This methods handles the form submit
+   *
+   * @param event
+   * @returns {boolean}
+   * @private
+   */
+  updatePartialForm(event) {
+    event.stopImmediatePropagation();
+
+    const updatedData = this.getUpdatedFormData();
+    if (updatedData !== null) {
+      this.postUpdatedData(updatedData);
+    } else {
+      // @todo: This is temporary we should probably use a nice modal instead, that said since the submit button is
+      //        disabled when no data has been modified it should never happen
+      alert('no fields updated');
+    }
+
+    return false;
+  }
+
+  /**
+   * Dynamically build a form with provided updated data and submit this "shadow" form
+   *
+   * @param updatedData {Object} Contains an object with all form fields to update indexed by query parameters name
+   */
+  postUpdatedData(updatedData) {
+    this.$productFormSubmitButton.prop('disabled', true);
+    const $updatedForm = this.$productForm.clone();
+    $updatedForm.empty();
+    $updatedForm.prop('class', '');
+    Object.keys(updatedData).forEach((fieldName) => {
+      $('<input>').attr({
+        name: fieldName,
+        type: 'hidden',
+        value: updatedData[fieldName],
+      }).appendTo($updatedForm);
+    });
+
+    $updatedForm.appendTo('body');
+    $updatedForm.submit();
+  }
+
+  /**
+   * Adapt the submit button state, as long as no data has been updated the button is disabled
+   */
+  updateSubmitButtonState() {
+    const updatedData = this.getUpdatedFormData();
+    this.$productFormSubmitButton.prop('disabled', updatedData === null);
+  }
+
+  /**
+   * Returns the updated data, only fields which are different from the initial page load
+   * are returned (token and method are added since they are required for a valid request).
+   *
+   * If no fields have been modified this method returns null.
+   *
+   * @returns {{}|null}
+   */
+  getUpdatedFormData() {
```

Comment:
Okay for now but in the end I'd prefer a model/store based implementation. It will make the form interaction much easier to manage.